librz/util/table.c

```diff
@@ -1339,11 +1372,12 @@ RZ_API void rz_table_visual_list(RzTable *table, RzList /*<RzListInfo *>*/ *list
 					info->pitv.addr,
 					b,
 					rz_itv_end(info->pitv),
-					(info->perm != -1) ? rz_str_rwx_i(info->perm) : "",
+					perm_str ? perm_str : "",
 					(info->extra) ? info->extra : "",
 					(info->name) ? info->name : "");
 			}
 			free(b);
+			free(perm_str);
 			i++;
 		}
```

Comment:
`perm_str` leaks if `!opts->va && info->perm != -1`, just move the `free(perm_str)`outside the if